### PR TITLE
fix: Revert to `legacy` format; fix code-generation to be more resiliant

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

**Review by steve8708 — APPROVED** [review]

File: packages/core/src/__tests__/__snapshots__/react.test.ts.snap [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default function MyBasicComponent(props) {
 
   return (
     <div>
-      <input value={name} onChange={(event) => setName(event.target.value)} />
+      <input onChange={(event) => setName(event.target.value)} />
```
Comment (steve8708) [inline packages/core/src/__tests__/__snapshots__/react.test.ts.snap:11]: something looks wrong here

File: packages/core/src/generators/react.ts [hunk 2]
```diff
@@ -160,7 +163,9 @@ export const blockToReact = (json: JSXLiteNode, options: ToReactOptions) => {
         str += ` ${BINDING_MAPPERS[key]}={${useBindingValue}} `;
       }
     } else {
-      str += ` ${key}={${useBindingValue}} `;
+      if (!isValidAttributeName(key)) {
```
Comment (steve8708) [inline packages/core/src/generators/react.ts:166]:
```suggestion
      if (isValidAttributeName(key)) {
```

this should fix the snapshots

Comment (mhevery) [reply packages/core/src/generators/react.ts:166]: fixed

File: packages/core/src/__tests__/__snapshots__/react.test.ts.snap (hunk 1 above)
Comment (mhevery) [reply packages/core/src/__tests__/__snapshots__/react.test.ts.snap:11]: Nice catch! fixed.

**Pull request merged**